vm/src/vm.rs

```diff
@@ -1537,29 +1577,76 @@ impl<'vm> Drop for ReprGuard<'vm> {
     }
 }
 
+pub struct Interpreter {
```

Comment:
does Interpreter matche to vm 1-to-1 relationship or is there any other usage of vm?
If it is 1-to-1, I don't feel like we really need a new type for that; we only need the new methods.

Reply:
The separate struct is so that you can't access the VM except through putting it in the VM stack, otherwise, it wouldn't be guaranteed that the VM would be in stack while using it

Reply:
In that case, will there be any difference if we put interpreters in VM stack instead of VMs?

Reply:
I initially did that, but then it was tricky passing an Interpreter to `initialize(&mut interp.vm, &interp)`. Definitely could make it happen though, especially if we're making init_importlib crate-private

Reply:
ah, got it. interface visibility is a good reason.

Reply:
This actually feels like a 1-to-n relationship as each thread has it's own vm.

Reply:
it seems each thread has its own vm and also its own interpreter. so still vm and interpreter is 1-to-1.

Reply:
I was originally thinking that Interpreter would cover all vms across threads for that interpreter instance; like it could have a thread-local value to hold the vm, and it could be Send+Sync and whichever thread you called it from it would get that thread's vm, but I'm not sure how actually useful that would be.